cocos/spine/assembler/simple.ts

```diff
@@ -141,70 +138,62 @@ function updateComponentRenderData (comp: Skeleton, batcher: Batcher2D): void {
     const ic = model.iCount as number;
-    const rd = comp.renderData!;
+    if (vc < 1 || ic < 1) return;
 
+    const rd = comp.renderData!;
     if (rd.vertexCount !== vc || rd.indexCount !== ic) {
         rd.resize(vc, ic);
         rd.indices = new Uint16Array(ic);
+        comp._vLength = vc * Float32Array.BYTES_PER_ELEMENT * floatStride;
+        comp._vBuffer = new Uint8Array(rd.chunk.vb.buffer, rd.chunk.vb.byteOffset, Float32Array.BYTES_PER_ELEMENT * rd.chunk.vb.length);
+        comp._iLength = Uint16Array.BYTES_PER_ELEMENT * ic;
+        comp._iBuffer = new Uint8Array(rd.indices.buffer);
     }
-    if (vc < 1 || ic < 1) return;
 
     const vbuf = rd.chunk.vb;
-    const vUint8Buf = new Uint8Array(vbuf.buffer, vbuf.byteOffset, Float32Array.BYTES_PER_ELEMENT * vbuf.length);
-
     const vPtr = model.vPtr;
-    const vLength = vc * Float32Array.BYTES_PER_ELEMENT * floatStride;
-    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
-    const vData = spine.wasmUtil.wasm.HEAPU8.subarray(vPtr, vPtr + vLength);
-    vUint8Buf.set(vData as TypedArray);
-
     const iPtr = model.iPtr;
     const ibuf = rd.indices!;
-    const iLength = Uint16Array.BYTES_PER_ELEMENT * ic;
-    // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
-    const iData = spine.wasmUtil.wasm.HEAPU8.subarray(iPtr, iPtr + iLength);
-    const iUint8Buf = new Uint8Array(ibuf.buffer);
-    iUint8Buf.set(iData as TypedArray);
+    const HEAPU8 = spine.wasmUtil.wasm.HEAPU8;
+
+    comp._vBuffer?.set(HEAPU8.subarray(vPtr, vPtr + comp._vLength), 0);
+    comp._iBuffer?.set(HEAPU8.subarray(iPtr, iPtr + comp._iLength), 0);
     const chunkOffset = rd.chunk.vertexOffset;
-    for (let i = 0; i < ic; i++) {
-        ibuf[i] += chunkOffset;
-    }
+    for (let i = 0; i < ic; i++) ibuf[i] += chunkOffset;
 
-    const meshes = model.getMeshes();
-    const count = meshes.size();
+    const data = model.getData();
+    const count = data.size();
     let indexOffset = 0;
     let indexCount = 0;
-    for (let i = 0; i < count; i++) {
-        const mesh = meshes.get(i);
-        const material = _getSlotMaterial(mesh.blendMode as number, comp);
-        const textureID = mesh.textureID as number;
-        indexCount = mesh.iCount;
+    for (let i = 0; i < count; i += 6) {
+        indexCount = data.get(i + 3);
+        const material = _getSlotMaterial(data.get(i + 4), comp);
+        const textureID: number = data.get(i + 5);
         comp.requestDrawData(material, textureID, indexOffset, indexCount);
         indexOffset += indexCount;
     }
+
     // if enableBatch apply worldMatrix
     if (comp.enableBatch) {
         const worldMat = comp.node.worldMatrix;
         let index = 0;
-        const tempVecPos = new Vec3(0, 0, 0);
         for (let i = 0; i < vc; i++) {
             index = i * floatStride;
             tempVecPos.x = vbuf[index];
             tempVecPos.y = vbuf[index + 1];
-            tempVecPos.z = 0;
```

Comment:
Don't need to reset the global variable `tempVecPos.z` to 0 ?

Reply:
Eventually this value will be set to vbuf[index+2] This will be one less assignment operation for now 
`
tempVecPos.y = vbuf[index + 1];
  tempVecPos.transformMat4(worldMat);
  vbuf[index] = tempVecPos.x;
  vbuf[index + 1] = tempVecPos.y;
  vbuf[index + 2] = 0;  
`

Reply:
Got it.